common/lib/xmodule/xmodule/modulestore/search.py

```diff
@@ -104,15 +104,23 @@ def find_path_to_course():
         # (e.g. sequential and videosequence) currently deal with this form of
         # representing nested positions. This needs to happen before jumping to a
         # module nested in more than one positional module will work.
+        masquerading_student = False
         if n > 3:
             position_list = []
+            if request.session.get('masquerade_settings') and \
+                    request.session.get('masquerade_settings').values()[0].role == 'student':
+                    masquerading_student = True
             for path_index in range(2, n - 1):
                 category = path[path_index].block_type
                 if category == 'sequential' or category == 'videosequence':
                     section_desc = modulestore.get_item(path[path_index])
                     # this calls get_children rather than just children b/c old mongo includes private children
                     # in children but not in get_children
-                    child_locs = [c.location for c in section_desc.get_children()]
+                    if request.user.is_staff and not masquerading_student:
```

Comment:
`request.user.is_staff` => Use `GlobalStaff` class to check the user is staff or not.